### PR TITLE
Reduxify

### PR DESCRIPTION
**Pull request by lyntco**
This branch adds a redux version of the component with extra reduxy things to include in your app.

it also removes css modules for now as it is not being used.

### CODE REVIEW

File: src/redux/reducer.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      const { name } = action.data;
+      if (!name) return state;
+      return state.merge({
+        [name]: { isOpen: false, selected: [] },
+      });
+    }
+    case C.SELECT_TOGGLE_OPEN: {
+      const { name, open } = action.data;
+      return state.merge({
+        [name]: (state.get(name) || fromJS({})).merge({ isOpen: open }),
```
Comment (danielmoi) [inline src/redux/reducer.js]: not sure what `[name]` is here... and the `fromJS({})` – are we clearing the entire state and setting it to `action.data.open` if there is no `name`?

File: src/redux/actions.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
+import C from './constants';
+
+const addSet = data => ({
+  type: C.SELECT_ADD_SET,
+  data,
+});
+
+const searchOptions = data => ({
+  type: C.SELECT_SEARCH_VALUES,
```
Comment (danielmoi) [inline src/redux/actions.js]: keep the constants same as action creator name?

File: src/SelectConnected.js [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  selectValue,
+  searchValues,
+} from './redux/actions';
+import SelectComponent, { propTypes, defaultProps } from './SelectComponent';
+
+export class Select extends Component {
+  onChange = value => () => {
+    const { name, selected } = this.props;
+    if (this.props.isMultipleSelect) {
+      if (selected.includes(value)) {
```
Comment (anupvarghese) [inline src/SelectConnected.js]: good idea to move this section to another function

File: src/SelectState.js [hunk 4]
```diff
@@ -44,6 +44,7 @@ class Select extends Component {
         identifier={this.props.identifier}
         label={this.props.label}
         options={this.state.options}
+        selected={this.state.selected || []}
```
Comment (anupvarghese) [inline src/SelectState.js]: probably a good idea to keep these as stateless functions and keep all the states in redux store

Comment (lyntco) [inline src/SelectState.js]: This component is for specific use of state in react, the other component is for redux store

File: src/SelectConnected.js (hunk 3 above)
Comment (lyntco) [inline src/SelectConnected.js]: :+1:

File: src/SelectConnected.js [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+    } else {
+      this.props.selectValue({ name, values: [value] });
+      this.props.toggleOpen({ name: this.props.name, open: false });
+    }
+  }
+
+  onSearch = text => () => {}
+
+  onToggleOpen = () => {
+    this.props.toggleOpen({ name: this.props.name, open: !this.props.isOpen });
```
Comment (danielmoi) [inline src/SelectConnected.js]: should we make it `isOpen`?

File: src/redux/reducer.js (hunk 1 above)
Comment (danielmoi) [inline src/redux/reducer.js]: is it because we are key-ing which select is being toggled?

Comment (lyntco) [inline src/redux/reducer.js]: we are indeed, so a Select with `name="testing"` will make the select object like this: 
``` 
{ 
  'testing': { 
    isOpen: false,
    selected: [],
  } 
}
```

File: src/redux/actions.js (hunk 2 above)
Comment (lyntco) [inline src/redux/actions.js]: yeah this is actually burning me now lol

**Pull request merged**